generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/Resolver.java

```diff
@@ -328,11 +333,37 @@ private boolean addArtifact(Rule dependency, String parent) {
       return false;
     }
 
+    updateVersion(artifactName, dependency);
     deps.put(artifactName, dependency);
     dependency.addParent(parent);
     return true;
   }
 
+  private void updateVersion(String artifactName, Rule dependency) {
+    VersionRange versionRange;
+    if (!restriction.containsKey(artifactName)) {
+      return;
+    }
+    String versionRestriction = restriction.get(artifactName);
+    try {
+      versionRange = VersionRange.createFromVersionSpec(versionRestriction);
+    } catch (InvalidVersionSpecificationException e) {
+      logger.warning(
+          "Error parsing version " + versionRestriction + ": " + e.getLocalizedMessage());
+      // So that this isn't logged over and over.
+      restriction.remove(artifactName);
+      return;
+    }
+    if (!versionRange.containsVersion(new DefaultArtifactVersion(dependency.version()))) {
+      try {
+        dependency.setVersion(
+            resolveVersion(dependency.groupId(), dependency.artifactId(), versionRestriction));
```

Comment:
Can you include a TODO or FIXME statement here? 

This resolveVersion is the incorrect one defined [here](https://github.com/bazelbuild/migration-tooling/pull/44/commits/17b568cbd53e4c67b754ee24c331fed51a81bc7c#diff-cc04e6e62b5271603d5f3061cf8fc024L102). Using that function ensures we are consistent about how we resolve versions/version ranges, but it is still incorrect. I think its important we make note of this as a reminder.

If you recall, we currently interpret "[1.0, )" as 1.0 rather than the highest available, and fail for things like "(3, 5)" which we shouldn't. This means we will erroneously receive InvalidArtifactCoordinateExceptions.